sql/parser/eval.go

```diff
@@ -448,6 +448,8 @@ type EvalContext struct {
 	NodeID        uint32
 	StmtTimestamp time.Time
 	TxnTimestamp  time.Time
+	Location      string
```

Comment:
Even if `time.LoadLocation` is slow, we could put a cache in front of it. I'd prefer to make this a `time.Location` and just resolve the session offset/location to a time zone in sql/executor.go.